packages/ui/src/ui/UIFactory.tsx

```diff
@@ -389,6 +390,16 @@ export interface UIFactory {
     getAclPermissionsSettings(): typeof PERMISSIONS_SETTINGS;
 
     onChytAliasSqlClick(params: {alias: string; cluster: string}): void;
+
+    getNavigationExtention(): {
```

Comment:
It looks like it is enough to describe additional tabs in declarative way, like:
```ts
type TabName = string;
type ExtraTab = {
    value: `extra_${TabName}`,
    title: string;
    component: React.ComponentType;
    position?: {before: TabName} | {after: TabName}
}
...
getNavigationExtraTabs(): undefined | Array<ExtraTab>;
```

Reply:
Done